homeassistant/components/alphaess/coordinator.py

```diff
@@ -0,0 +1,48 @@
+"""Coordinator for AlphaEss integration."""
+
+from datetime import timedelta
+import logging
+
+from alphaess import alphaess
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .const import DOMAIN
+
+_LOGGER: logging.Logger = logging.getLogger(__package__)
+SCAN_INTERVAL = timedelta(minutes=5)
+PLATFORMS = [Platform.SENSOR]
+
+
+class AlphaESSDataUpdateCoordinator(DataUpdateCoordinator):
+    """Class to manage fetching data from the API."""
+
+    def __init__(self, hass: HomeAssistant, client: alphaess.alphaess) -> None:
+        """Initialize."""
+        super().__init__(hass, _LOGGER, name=DOMAIN, update_interval=SCAN_INTERVAL)
+        self.api = client
+        self.update_method = self._async_update_data
+
+    async def _async_update_data(self):
+        """Update data via library."""
+        try:
+            _LOGGER.info("Trying to query AlphaESS api data")
+            return await self.api.getdata()
+        except Exception as exception:
```

Comment:
Remove broad except, DataUpdateCoordinator already handles those broad excepts.

Reply:
Fixed thank you - I know separately catch client response errors and connection errors and leave all else to the DataUpdateCoordinator